components/Container/index.tsx

```diff
@@ -0,0 +1,292 @@
+'use client'
+
+import React, {
+  CSSProperties,
+  useState,
+  useEffect,
+  useRef,
+} from 'react';
+
+import {
+  Button,
+} from 'react-aria-components';
+
+import { DmpIcon } from '@/components/Icons';
+import {
+  useResponsive,
+  getSizeByName,
+} from '@/hooks/responsive';
+
+import './containers.scss';
+
+
+type DirectionType =
+  | null
+  | "left"
+  | "right";
+
+// Extend the CSSProps so that we can actually add css variables to tags.
+type CustomCSSProperties = CSSProperties & Record<string, string>;
+
+/**
+ * This is the base layout component. Our custom layout components should
+ * inherrit from this one. So that we have some common functionality for all
+ * layout containers.
+ */
+interface LayoutContainerProps extends React.HTMLAttributes<HTMLDivElement> {
+  style?: CustomCSSProperties,
+}
+
+export const LayoutContainer: React.FC<LayoutContainerProps> = ({
+  id,
+  className,
+  style,
+  children,
+}) => {
+  return (
+    <>
+      <div
+        id={id}
+        className={`layout-container ${className}`}
+        style={style}
+      >
+        {children}
+      </div>
+    </>
+  )
+}
+
+
+/**
+ * ContentContainer is the base component that is meant to hold content
+ * inside a LayoutContainer.
+ */
+interface ContentContainerProps extends React.HTMLAttributes<HTMLDivElement> {
+  style?: CustomCSSProperties,
+}
+
+export const ContentContainer: React.FC<ContentContainerProps> = ({
+  children,
+  id = '',
+  className = '',
+  style,
```

Comment:
Will styles for ContentContainer ever vary, or will they always be consistent across pages?

Reply:
I removed the "style" property. As @jupiter007 rightly points out we don't really need to change this property ever.